components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java

```diff
@@ -2155,6 +2188,50 @@ private void parseOpenIDConnectConfig(OMElement oauthConfigElem) {
         }
     }
 
+    /**
+     * Parse supported encryption algorithms set and add them to supportedIdTokenEncryptionAlgorithms.
+     *
+     * @param algorithms OMElement of supported algorithms.
+     */
+    private void parseSupportedIdTokenEncryptionAlgorithms(OMElement algorithms) {
+
+        if (algorithms == null) {
+            return;
+        }
+
+        Iterator iterator = algorithms.getChildrenWithLocalName(ConfigElements.SUPPORTED_ID_TOKEN_ENCRYPTION_ALGORITHM);
+        if (iterator != null) {
+            for (; iterator.hasNext(); ) {
```

Comment:
better to use while loop

Reply:
Fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/960